types/api/query/chain.go

```diff
@@ -12,13 +13,36 @@ type RequestByUrl struct {
 	Url types.String
 }
 
+type RequestByUrlAndKey struct {
+	RequestByUrl
+	Key string
+}
+
+type KeyPageResponse struct {
+	KeyPage string `json:"key_page"`
+	Indices []int  `json:"indices"`
+}
+
+type ResponseKeyPageIndices struct {
+	KeyBook        string             `json:"key_book"`
+	KeyPageIndices []*KeyPageResponse `json:"key_page_indices"`
+}
```

Comment:
This needs to be moved to `internal/api/v2/types.yml`. Types specific to API v2 do not belong here.